src/lib/Libifl/pbsD_msgjob.c

```diff
@@ -227,12 +227,13 @@ char *extend;
 		int i;
 		struct pbs_client_thread_connect_context *con;
 		char nd_ct_selstr[20];
-		char *endptr = NULL;
+		char *endptr = NULL, *extend_dup = strdup(extend), *num_only;
```

Comment:
Need to check the return value of "strdup()" here, otherwise, you'll crash on the strtok(extend_dup,...) next.

Reply:
https://github.com/PBSPro/pbspro/pull/1452/commits/122ff5230827085550ab21708ca48da405d844a6 fixes it. thanks